packages/flutter/test/widgets/automatic_keep_alive_test.dart

```diff
@@ -518,3 +519,48 @@ class _AlwaysKeepAliveState extends State<_AlwaysKeepAlive> with AutomaticKeepAl
     );
   }
 }
+/// Theses classes are here to make sure that [KeepAlive] is dissociate from [RenderSliverMultiBoxAdaptor].
```

Comment:
A more valuable test here would be a simple implementation of KeepAlive Sliver that is not a MultiBoxAdaptor.

You might take, for example, whatever you're doing that led to this and pare it down and include it as a test here

Reply:
Yes you're right, this test doesn't help to understand why this is needed.
The issue I'm facing, is that I have to find a scenario with a minimum of code (what led me to this is has a lot of code).

Reply:
It's ok if you create a separate file or files for this, and it's also ok if the code isn't really meant to be used for anything but proving we don't re-entangle the keep alive logic with SliverMultiBoxAdaptor

Reply:
Ok, I will give a try this weekend then 😊

Reply:
There are really too much classes to create if I want a complete example. I'm not sure it will help anybody if I copy/paste the classes from https://github.com/letsar/flutter_staggered_grid_view/tree/master/lib/src/rendering.
I had to create all theses classes to have a RenderSliverMultiBoxAdaptor that does not maintain a doubly-linked list of its children (because I wanted random access). This lead to me to think, that maybe it would be better if the Flutter SDK provides another RenderSliverMultiBoxAdaptor that delegates how the children are stored.

Reply:
What it helps is make sure that we don't regress this functionality in the future.  We want to be sure there's a test that exercises this functionality in a (at least somewhat) realistic way, even if that way isn't something we're directly integrating into the framework.  That way, someone looking to simplify or enhance code won't accidentally back these changes out (at least not without breaking tests).

We have some precedent for this [here](https://github.com/flutter/flutter/blob/master/packages/flutter/test/widgets/linked_scroll_view_test.dart) - that functionality isn't something that we necessarily want or need in the framework right now, but we want to make sure we don't break the ability to do it in the future.

Does that help?

Reply:
Yes I understand and share your point of view :-). I need to find a simpler test that what I was going to do.
In the meantime maybe I also need to find a way to provider another `RenderSliverMultiBoxAdaptor `that delegates how the children are stored, but this is maybe too much work...
Thanks for the link :-).